packages/core/src/external/quest/fhir/service-request.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  CodeableConcept,
+  Coding,
+  Identifier,
+  Practitioner,
+  Reference,
+  ServiceRequest,
+} from "@medplum/fhirtypes";
+import { uuidv7 } from "@metriport/shared/util/uuid-v7";
+import { ResponseDetail } from "../schema/response";
+import { getPractitionerReference } from "./practitioner";
+import { getQuestDataSourceExtension } from "./shared";
+
+export function getServiceRequest(
+  detail: ResponseDetail,
+  { requestingPractitioner }: { requestingPractitioner: Practitioner }
+): ServiceRequest {
+  const identifier = getServiceRequestIdentifier(detail);
+  const requester = getPractitionerReference(requestingPractitioner);
+  const code = getServiceRequestCoding(detail);
+  const extension = [getQuestDataSourceExtension()];
+
+  return {
+    resourceType: "ServiceRequest",
+    id: uuidv7(),
+    status: "completed",
+    intent: "original-order",
+    ...(code ? { code } : {}),
+    ...(identifier ? { identifier } : {}),
+    requester,
+    extension,
+  };
+}
+
+export function getServiceRequestReference(
+  serviceRequest: ServiceRequest
+): Reference<ServiceRequest> {
+  return {
+    reference: `ServiceRequest/${serviceRequest.id}`,
+  };
+}
+
+function getServiceRequestIdentifier(detail: ResponseDetail): Identifier[] | undefined {
+  if (!detail.requisitionNumber) return undefined;
+  return [
+    {
+      system: "http://questdiagnostics.com/requisition-number",
+      value: detail.requisitionNumber,
+    },
+  ];
+}
+
+function getServiceRequestCoding(detail: ResponseDetail): CodeableConcept | undefined {
```

Comment:
do you know what coding system these profile + order codes are using? Are these coding systems right?

Reply:
It says these should be snomed codes.

Reply:
Agree that these should not be order codes and be medical codes.